src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -347,5 +347,127 @@ private static bool IsMultidimensionalInitializer(ImmutableArray<BoundExpression
 
             return inits.Length != 0 && inits[0].Kind == BoundKind.ArrayInitialization;
         }
+
+        private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExpression wrappedExpression, bool used, bool inPlace)
+        {
+            ImmutableArray<byte> data = default;
+            int elementCount = -1;
+            TypeSymbol elementType = null;
+
+            if (!_module.SupportsPrivateImplClass)
+            {
+                return false;
+            }
+
+            var ctor = ((MethodSymbol)this._module.Compilation.GetWellKnownTypeMember(WellKnownMember.System_ReadOnlySpan_T__ctor));
+            if (ctor == null)
+            {
+                return false;
+            }
+
+            if (wrappedExpression is BoundArrayCreation ac)
```

Comment:
Perhaps invert this logic:
`if (wrappedExpression.Kind() != BoundKind.ArrayCreation) { return false; }`
Then `elementCount`, `data` and `elementType` can be declared once they are needed and initialized.

Reply:
The code started with support for arrays and string. Now, because of endiannes it was reduced to supporting just array, but once we have a necessary runtime helper, we would definitely want to support strings.
Then bailing out on negative condition would not be convenient. 
With all other parts being equal, positive check is easier to scale to multiple types in the future, that is why we use a positive check here.


---
In reply to: [172329090](https://github.com/dotnet/roslyn/pull/24621#discussion_r172329090) [](ancestors = 172329090)